components/IconSearch.js

```diff
@@ -0,0 +1,47 @@
+import { useState } from 'react'
+import { Combobox } from '@headlessui/react'
+import * as FaIcons from "react-icons/fa";
+import * as SiIcons from "react-icons/si";
+import getIcon from './Icon';
+
+function IconSearch({selectedIcon, handleSelectedIcon}) {
+  const [query, setQuery] = useState('');
+  
+  const popularIcons = ["FaGithub", "FaTwitter", "FaLinkedin", "FaGit"];
+  const icons = [];
+  
+  Object.keys(FaIcons).forEach((key) => {
+    icons.push(key);
+  });
+
+  Object.keys(SiIcons).forEach((key) => {
+    icons.push(key);
+  });
```

Comment:
This can be simplified and more efficient with 

```suggestion
const icons = [...Object.keys(FaIcons), ...Object.keys(SiIcons)];
```